bson/_cbsonmodule.c

```diff
@@ -281,13 +283,40 @@ static PyObject* datetime_from_millis(long long millis) {
     struct TM timeinfo;
     cbson_gmtime64_r(&seconds, &timeinfo);
 
-    return PyDateTime_FromDateAndTime(timeinfo.tm_year + 1900,
+    PyObject* datetime_hold = PyDateTime_FromDateAndTime(timeinfo.tm_year + 1900,
                                       timeinfo.tm_mon + 1,
                                       timeinfo.tm_mday,
                                       timeinfo.tm_hour,
                                       timeinfo.tm_min,
                                       timeinfo.tm_sec,
                                       microseconds);
+    if(!datetime_hold) {
+        PyObject *etype, *evalue, *etrace;
+
+        /*
+        * Calling _error clears the error state, so fetch it first.
+        */
+        PyErr_Fetch(&etype, &evalue, &etrace);
+
+        /* Only add addition error message on ValueError exceptions. */
+        if (PyErr_GivenExceptionMatches(etype, PyExc_ValueError)) {
+            if (evalue) {
+                PyObject* err_msg = PyObject_Repr(evalue);
+                if(err_msg) {
+                    PyObject* appendage = PyUnicode_FromString(" (Use CodecOptions(datetime_conversion=DATETIME_AUTO) or MongoClient(datetime_conversion='DATETIME_AUTO'))");
```

Comment:
We need to error check the results of PyUnicode_FromString and PyUnicode_Concat.